packages/back-end/src/controllers/datasources.ts

```diff
@@ -513,6 +514,35 @@ export async function putDataSource(
     if (params) {
       const integration = getSourceIntegrationObject(context, datasource);
       mergeParams(integration, params);
+
+      // TODO: Should the validation be done here or somewhere else?
```

Comment:
Not here. This method runs `testConnection` to verify the params before saving. So as long as running the query fails, you don't need any additional validation here.

Reply:
Removed and the error message was fixed by adding a proper try/catch when decrypting the key.